migration-engine/connectors/migration-connector/src/imperative_migrations_persistence.rs

```diff
@@ -15,6 +15,35 @@ pub trait ImperativeMigrationsPersistence: Send + Sync {
     /// migration persistence. If not, return a DatabaseSchemaNotEmpty error.
     async fn initialize(&self) -> ConnectorResult<()>;
 
+    /// Implementation in the connector for the core's MarkMigrationApplied
+    /// command. See the docs there. Note that the started_at and finished_at
+    /// for the migration should be the same.
+    ///
+    /// Connectors should implement mark_migration_applied_impl to avoid doing
+    /// the checksuming themselves.
+    async fn mark_migration_applied(&self, migration_name: &str, script: &str) -> ConnectorResult<String> {
+        self.mark_migration_applied_impl(migration_name, script, &checksum(script))
+            .await
+    }
+
+    /// Implementation in the connector for the core's MarkMigrationApplied
+    /// command. See the docs there. Note that the started_at and finished_at
+    /// for the migration should be the same.
+    async fn mark_migration_applied_impl(
+        &self,
+        migration_name: &str,
+        script: &str,
+        checksum: &str,
+    ) -> ConnectorResult<String>;
+
+    /// Mark the failed instances of the migration in the persistence as rolled
+    /// back, so they will be ignored by the engine in the future.
+    async fn mark_migration_rolled_back(&self, migration_name: &str) -> ConnectorResult<()>;
```

Comment:
This is not used anymore afaik, since we're using the by_id method below

Reply:
Deleting this :+1: